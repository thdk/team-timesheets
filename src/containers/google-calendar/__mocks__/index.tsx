import React from "react";

export const GoogleCalendarEvents = () => <>GoogleCalendarEvents</>;