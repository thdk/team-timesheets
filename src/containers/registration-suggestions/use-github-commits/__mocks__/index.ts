export const useGithubCommits = jest.fn().mockReturnValue({
    data: [],
    hasData: false,
});
