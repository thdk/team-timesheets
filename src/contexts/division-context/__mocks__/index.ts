export const useDivisionStore = jest.fn().mockReturnValue({
    userDivisions: [],
});
