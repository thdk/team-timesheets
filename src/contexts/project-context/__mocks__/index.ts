export const useProjectStore = jest.fn().mockReturnValue({
    projects: [],
    collection: {
        isFetched: true,
        docs: [],
    },
    activeProjects: [],
});
