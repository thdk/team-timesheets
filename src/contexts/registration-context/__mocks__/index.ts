export * from "./use-registration-store";
