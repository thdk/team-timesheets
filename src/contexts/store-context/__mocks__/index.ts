export const useStore = jest.fn();
