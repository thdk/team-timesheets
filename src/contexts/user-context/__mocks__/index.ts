export const useUserStoreMock = {
    authenticatedUser: {
        email: "foobar@email.com",
        name: "Foobar",
        roles: {
            admin: true,
        },
    },
    divisionUser: {
        email: "foobar@email.com",
        name: "Foobar",
        roles: {
            admin: true,
        },
    },
    usersCollection: {
        isFetched: true,
    },
    divisionUsersCollection: {
        isFetched: true,
    },
    updateDivisionUser: jest.fn(),
} as ReturnType<typeof useUserStore>;

export const useUserStore = jest.fn().mockReturnValue(useUserStoreMock);
