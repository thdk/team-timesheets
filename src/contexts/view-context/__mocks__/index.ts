import moment from "moment";

export const useViewStore = jest.fn().mockReturnValue({
    isDrawerOpen: true,
    moment: moment(new Date(2020, 2 ,22)),
    toggleSelection: jest.fn(),
    selection: new Map(),
});
