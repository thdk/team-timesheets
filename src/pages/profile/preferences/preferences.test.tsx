import React from "react";
import { render } from "@testing-library/react";
import { Preferences } from ".";
import { useTasks } from "../../../contexts/task-context";
import { useClients } from "../../../contexts/client-context";
import { ITask, IClient } from "../../../../common";
import { useUserStore } from "../../../contexts/user-context";
import { useUserStoreMock } from "../../../contexts/user-context/__mocks__";

jest.mock('../../../contexts/user-context');
jest.mock('../../../contexts/auth-context');
jest.mock('../../../contexts/task-context');
jest.mock('../../../contexts/client-context');
jest.mock('../../../contexts/project-context');

jest.mock('./task-preferences', jest.fn().mockReturnValue(({
    TaskPreferences: () => <>TaskPreferences</>
})));
jest.mock('./client-preferences', jest.fn().mockReturnValue(({
    ClientPreferences: () => <>ClientPreferences</>
})));

jest.mock('./project-preferences', jest.fn().mockReturnValue(({
    ProjectPreferences: () => <>ProjectPreferences</>
})));

const resetMocks = () => {
    (useTasks as jest.Mock<ReturnType<typeof useTasks>>).mockReturnValue({
        tasks: [] as ITask[],
        collection: {
            isFetched: true,
        },
    } as ReturnType<typeof useTasks>);

    (useClients as jest.Mock<ReturnType<typeof useClients>>).mockReturnValue({
        clients: [] as IClient[],
        clientsCollection: {
            isFetched: true,
        },
    } as ReturnType<typeof useClients>);

    (useUserStore as jest.Mock<ReturnType<typeof useUserStore>>).mockReturnValue(
        useUserStoreMock
    );
}

beforeEach(resetMocks);
afterAll(resetMocks);

describe("Preferences", () => {
    it("should display a message when no clients or tasks has been added", () => {
        const { asFragment } = render(
            <Preferences />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it("should display task preferences", () => {
        (useTasks as jest.Mock<ReturnType<typeof useTasks>>).mockReturnValue({
            tasks: [
                {
                    id: "task-1",
                    name: "Task 1 ",
                },
                {
                    id: "task-2",
                    name: "Task 2 ",
                    icon: "people",
                },
                {
                    id: "task-3",
                    name: "Task 3 ",
                    icon: "code",
                },
            ],
            collection: {
                isFetched: true,
            }
        } as ReturnType<typeof useTasks>);

        const { asFragment } = render(
            <Preferences />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it("should display client preferences", () => {
        (useClients as jest.Mock<ReturnType<typeof useClients>>).mockReturnValue({
            clients: [{
                id: "client-1",
            }],
            clientsCollection: {
                isFetched: true,
            }
        } as ReturnType<typeof useClients>);

        const { asFragment } = render(
            <Preferences />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it("should not render when unauthenticated", () => {
        (useUserStore as jest.Mock<ReturnType<typeof useUserStore>>).mockReturnValue({
            divisionUser: undefined,
        } as ReturnType<typeof useUserStore>);

        const { asFragment } = render(
            <Preferences />
        );

        expect(asFragment()).toMatchSnapshot();
    })
});
