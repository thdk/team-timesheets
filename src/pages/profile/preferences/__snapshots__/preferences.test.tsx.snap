// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preferences should display a message when no clients or tasks has been added 1`] = `
<DocumentFragment>
  <div
    class="box"
  >
    <h3
      class="mdc-typography--subtitle1"
    >
      Your division has not been fully configured yet. Add some tasks from the settings page or ask you division manager to set this up.
    </h3>
  </div>
</DocumentFragment>
`;

exports[`Preferences should display client preferences 1`] = `
<DocumentFragment>
  <div
    class="box"
  >
    TaskPreferencesClientPreferences
    <h3
      class="mdc-typography--subtitle1"
    >
      Select number of recent projects
    </h3>
    <div
      class="formField"
    >
      <div
        class="mdc-select mdc-select--outlined"
        role="listbox"
      >
        <div
          class="mdc-select__anchor"
        >
          <i
            class="mdc-select__dropdown-icon"
          />
          <div
            aria-disabled="false"
            aria-haspopup="listbox"
            class="mdc-select__selected-text"
            role="button"
            tabindex="-1"
          >
            5
          </div>
          <div
            class="mdc-notched-outline mdc-notched-outline--upgraded"
            notch="0"
          >
            <div
              class="mdc-notched-outline__leading"
            />
            <div
              class="mdc-notched-outline__notch"
            >
              <span
                class="mdc-floating-label mdc-floating-label--float-above"
                for="select-# of recent projects"
                style="transition-duration: 0s;"
              >
                # of recent projects
              </span>
            </div>
            <div
              class="mdc-notched-outline__trailing"
            />
          </div>
          <select
            class="rmwc-select__native-control "
            name="recentProjects"
            role="combobox"
            tabindex="0"
          >
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
            <option
              value="7"
            >
              7
            </option>
            <option
              value="8"
            >
              8
            </option>
            <option
              value="9"
            >
              9
            </option>
            <option
              value="10"
            >
              10
            </option>
            <option
              value="11"
            >
              11
            </option>
            <option
              value="12"
            >
              12
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Preferences should display task preferences 1`] = `
<DocumentFragment>
  <div
    class="box"
  >
    TaskPreferencesClientPreferences
    <h3
      class="mdc-typography--subtitle1"
    >
      Select number of recent projects
    </h3>
    <div
      class="formField"
    >
      <div
        class="mdc-select mdc-select--outlined"
        role="listbox"
      >
        <div
          class="mdc-select__anchor"
        >
          <i
            class="mdc-select__dropdown-icon"
          />
          <div
            aria-disabled="false"
            aria-haspopup="listbox"
            class="mdc-select__selected-text"
            role="button"
            tabindex="-1"
          >
            5
          </div>
          <div
            class="mdc-notched-outline mdc-notched-outline--upgraded"
            notch="0"
          >
            <div
              class="mdc-notched-outline__leading"
            />
            <div
              class="mdc-notched-outline__notch"
            >
              <span
                class="mdc-floating-label mdc-floating-label--float-above"
                for="select-# of recent projects"
                style="transition-duration: 0s;"
              >
                # of recent projects
              </span>
            </div>
            <div
              class="mdc-notched-outline__trailing"
            />
          </div>
          <select
            class="rmwc-select__native-control "
            name="recentProjects"
            role="combobox"
            tabindex="0"
          >
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
            <option
              value="7"
            >
              7
            </option>
            <option
              value="8"
            >
              8
            </option>
            <option
              value="9"
            >
              9
            </option>
            <option
              value="10"
            >
              10
            </option>
            <option
              value="11"
            >
              11
            </option>
            <option
              value="12"
            >
              12
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Preferences should not render when unauthenticated 1`] = `<DocumentFragment />`;
